storage: BenchmarkMVCCExportToSST fails on numRangeKeys

To reproduce:
```
./dev bench pkg/storage -f=BenchmarkMVCCExportToSST --ignore-cache --bench-time=1x --verbose --stream-output --timeout 10m --test-args="-test.failfast"
```

Failure:
```
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%/exportAllRevisions=false
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%/exportAllRevisions=false-10 	       1	 295958333 ns/op
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%/exportAllRevisions=true
    bench_test.go:1684:
        	Error Trace:	/private/var/tmp/_bazel_herko/7349b8c7c2d76abd184572c7559a04c3/sandbox/darwin-sandbox/20/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/bench_test.go:1684
        	            				/private/var/tmp/_bazel_herko/7349b8c7c2d76abd184572c7559a04c3/sandbox/darwin-sandbox/20/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/bench_test.go:166
        	            				/private/var/tmp/_bazel_herko/7349b8c7c2d76abd184572c7559a04c3/sandbox/darwin-sandbox/20/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/benchmark.go:193
        	            				/private/var/tmp/_bazel_herko/7349b8c7c2d76abd184572c7559a04c3/sandbox/darwin-sandbox/20/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/benchmark.go:233
        	            				/private/var/tmp/_bazel_herko/7349b8c7c2d76abd184572c7559a04c3/sandbox/darwin-sandbox/20/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/asm_arm64.s:1172
        	Error:      	"198" is not less than or equal to "100"
        	Test:       	BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%/exportAllRevisions=true
--- FAIL: BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%/exportAllRevisions=true
--- FAIL: BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=100%
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=50%
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=50%/exportAllRevisions=false
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=50%/exportAllRevisions=false-10  	       1	 403153458 ns/op
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=50%/exportAllRevisions=true
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=50%/exportAllRevisions=true-10   	       1	1041253000 ns/op
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=10%
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=10%/exportAllRevisions=false
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=10%/exportAllRevisions=false-10  	       1	 398654292 ns/op
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=10%/exportAllRevisions=true
BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02/perc=10%/exportAllRevisions=true-10   	       1	 512182750 ns/op
--- FAIL: BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100/numRangeKeys=[0_1_100]#02
--- FAIL: BenchmarkMVCCExportToSST/numKeys=65536/numRevisions=100
--- FAIL: BenchmarkMVCCExportToSST/numKeys=65536
```

Note: 
```
Error:      	"198" is not less than or equal to "100"
```

Jira issue: CRDB-24004

Epic CRDB-20293

--- a/pkg/storage/bench_test.go
+++ b/pkg/storage/bench_test.go
@@ -152,7 +152,7 @@ func BenchmarkMVCCExportToSST(b *testing.B) {
 			for _, numRevision := range numRevisions {
 				b.Run(fmt.Sprintf("numRevisions=%d", numRevision), func(b *testing.B) {
 					for _, numRangeKey := range numRangeKeys {
-						b.Run(fmt.Sprintf("numRangeKeys=%d", numRangeKeys), func(b *testing.B) {
+						b.Run(fmt.Sprintf("numRangeKeys=%d", numRangeKey), func(b *testing.B) {
 							for _, perc := range []int{100, 50, 10} {
 								if numRevision == 1 && perc != 100 {
 									continue // no point in incremental exports with 1 revision
@@ -1666,7 +1666,7 @@ func runMVCCExportToSST(b *testing.B, opts mvccExportToSSTOpts) {
 	it.SeekGE(MakeMVCCMetadataKey(roachpb.LocalMax))
 	require.NoError(b, err)
 	var n int // points
-	var r int // range key stacks
+	var r int // range keys (within stacks)
 	for {
 		ok, err := it.Valid()
 		require.NoError(b, err)
@@ -1678,7 +1678,7 @@ func runMVCCExportToSST(b *testing.B, opts mvccExportToSSTOpts) {
 			n++
 		}
 		if hasRange && it.RangeKeyChanged() {
-			r++
+			r += it.RangeKeys().Len()
 		}
 		it.Next()
 	}
@@ -1686,7 +1686,7 @@ func runMVCCExportToSST(b *testing.B, opts mvccExportToSSTOpts) {
 	// Should not see any rangedel stacks if startTS is set.
 	if opts.numRangeKeys > 0 && startWall == 0 && opts.exportAllRevisions {
 		require.NotZero(b, r)
-		require.LessOrEqual(b, r, opts.numRangeKeys)
+		require.GreaterOrEqual(b, r, opts.numRangeKeys)
 	} else {
 		require.Zero(b, r)
 	}

